src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbs.tsx

```diff
@@ -0,0 +1,368 @@
+
+    if (isCollapsed && filteredCollapsedBreadcrumbs.length > MAX_CRUMBS_WHEN_COLLAPSED) {
+      filteredCollapsedBreadcrumbs = filteredCollapsedBreadcrumbs.slice(
+        -MAX_CRUMBS_WHEN_COLLAPSED
+      );
+    }
+
+    return {
+      filteredCollapsedBreadcrumbs,
+      collapsedQuantity: filteredBreadcrumbs.length - filteredCollapsedBreadcrumbs.length,
+    };
+  };
+
+  handleFilter = (breadcrumbFilterGroups: BreadcrumbFilterGroups) => () => {
+    //types
+    const breadcrumbFilterGroupTypes = breadcrumbFilterGroups.filter(
+      breadcrumbFilterGroup => breadcrumbFilterGroup.groupType === 'type'
+    );
+
+    // TODO(Priscila): implement levels
+    // const breadcrumbFilterGroupLevels = breadcrumbFilterGroups
+    //   .filter(breadcrumbFilterGroup => breadcrumbFilterGroup.groupType === 'level')
+    //   .map(breadcrumbFilterGroup => breadcrumbFilterGroup.type);
+
+    this.setState({
+      filteredBreadcrumbs: this.state.breadcrumbs.filter(breadcrumb => {
+        const foundBreadcrumbFilterData = breadcrumbFilterGroupTypes.find(
+          crumbFilterData => crumbFilterData.type === breadcrumb.type
+        );
+        if (foundBreadcrumbFilterData) {
+          return foundBreadcrumbFilterData.isChecked;
+        }
+
+        return false;
+      }),
+      breadcrumbFilterGroups,
+    });
+  };
+
+  handleFilterBySearchTerm = (value: string) => {
+    const {filteredBreadcrumbsByCustomSearch} = this.state;
+
+    const searchTerm = value.toLocaleLowerCase();
+
+    const filteredBreadcrumbs = filteredBreadcrumbsByCustomSearch.filter(
+      item =>
+        !!['category', 'message', 'level', 'timestamp'].find(prop => {
+          const searchValue = item[prop];
+          if (searchValue) {
+            return searchValue.toLowerCase().indexOf(searchTerm) !== -1;
+          }
+          return false;
+        })
+    );
+
+    this.setState({
+      searchTerm,
+      filteredBreadcrumbs,
+    });
+  };
+
+  handleCollapseToggle = () => {
+    this.setState(prevState => ({
+      isCollapsed: !prevState.isCollapsed,
+    }));
+  };
+
+  handleCleanSearch = () => {
+    this.setState({
+      searchTerm: '',
+      isCollapsed: true,
+    });
+  };
+
+  render() {
+    const {event, type} = this.props;
+    const {breadcrumbFilterGroups, searchTerm} = this.state;
+
+    const {
+      collapsedQuantity,
+      filteredCollapsedBreadcrumbs,
+    } = this.getCollapsedCrumbQuantity();
+
+    return (
+      <EventDataSection
+        type={type}
+        title={
+          <h3>
+            <GuideAnchor target="breadcrumbs" position="bottom">
+              {t('Breadcrumbs')}
+            </GuideAnchor>
+          </h3>
+        }
+        actions={
+          <Search>
+            <BreadcrumbFilter
+              onFilter={this.handleFilter}
+              filterGroups={breadcrumbFilterGroups}
+            />
+            <StyledSearchBar
+              placeholder={t('Search breadcrumbs...')}
```

Comment:
Should we use a unicode ellipsis here? \u2026

Reply:
Please, why would it be better?